data_scrapers/san_francisco_county.py

```diff
@@ -4,101 +4,238 @@
-
-def get_ethnicity_json() -> Dict:
-    """ group data by ethnicity"""
-    ethnicity_query = '?$select=ethnicity, sum(confirmed_cases)&$group=ethnicity'
-    return get_json(race_ethnicity_url, ethnicity_query)
-
-# Confirmed Cases and Deaths by Date and Transmission
-def get_cases_and_deaths() -> Dict:
-    """Get total confirmed cases and deaths by summing over transmission table.
-    Note that total confrimed cases excludes deaths."""
-    query = '?$select=case_disposition, sum(case_count)&$group=case_disposition'
-    return get_json(transmission_url + query)
-
-def get_date_transmission_json() -> Dict:
-    """Get cases by date, transmission, and disposition; order by date ascending."""
-    # date_order_query = '?$order=date'
-    return get_json(transmission_url)
-
-def get_transmission_json() -> Dict:
-    """Group data by transmission category"""
-    transmission_query = '?$select=transmission_category, sum(case_count)&$group=transmission_category'
-    return get_json(transmission_url, transmission_query)
-
-# COVID+ patients from all SF hospitals in ICU vs. acute care, by date
-def get_hospitalization_json() -> Dict:
-    """Get number of COVID+ patients by bed category (ICU or regular), order by date ascending."""
-    date_order_query = '?$order=reportdate'
-    return get_json(hospitalizations_url + date_order_query)
-
-def get_icu_beds() -> Dict:
-    """group data by bed type: ICU or regular"""
-    icu_query = '?$select=dphcategory, sum(patientcount)&$group=dphcategory'
-    return get_json(hospitalizations_url + icu_query)
-
-# Daily count of tests with count and percent of positive tests
-def get_tests() -> Dict:
-    """Get tests by day, order by date ascending"""
-    date_order_query = '?$order=result_date'
-    return get_json(tests_url + date_order_query)
-
-def get_test_totals() -> Dict:
-    """Get total tests and total number of positive tests, to date"""
-    total_test_query = '?$select=sum(tests) as tests'
-    tests = get_json(tests_url + total_test_query)
-    total_positives_query = '?$select=sum(pos) as total_positives'
-    positives = get_json(tests_url + total_positives_query)
-    tests[0].update(positives[0])
-    return tests
+    response = requests.get(race_ethnicity_url)
+    response.raise_for_status()
+    # Dict of target_label : source_label for re-keying.
+    # Note: Native_Amer not currently reported
+    RACE_ETH_KEYS = {"Latinx_or_Hispanic": "Hispanic or Latino", "Asian": "Asian", "African_Amer": "Black or African American",
+                 "White": "White", "Pacific_Islander": "Native Hawaiian or Other Pacific Islander", "Multiple_Race": "Multiple Race",
+                 "Other": "Other", "Unknown": "Unknown"}
+    data = response.json()
+    # re-key and aggregate to flatten race x ethnicity
+    race_eth_data: Dict[str,int] = { k:0 for k in RACE_ETH_KEYS.keys() } # initalize all categories to 0 for aggregating
+
+    for item in data: # iterate through all race x ethnicity objects
```

Comment:
I'd like some more eyeballs on this logic. I'm getting numbers close to how SF county flattens race and ethnicity on their dashboard, but not exact. The discrepancy might be the lag between the data update and the dashboard update, but then you would expect our numbers to be slightly higher. However, they are slightly lower for Hispanic or Latino, and Other. I think Jason Lally mentioned that they are revisiting their data categories. Perhaps some "Unknown" and "Other" patients are being re-classified, explaining the drop in numbers in those categories today compared to the dashboard numbers. 

Here's what this for loop produces:
```
   "race_eth": {
            "Latinx_or_Hispanic": 828,
            "Asian": 257,
            "African_Amer": 98,
            "White": 343,
            "Pacific_Islander": 23,
            "Multiple_Race": 10,
            "Other": 99,
            "Unknown": 336
        }
```

Here's what the dashboard looks like (updated 5/12):
![Screen Shot 2020-05-13 at 8 53 20 AM](https://user-images.githubusercontent.com/12928553/81835590-485b5700-94f7-11ea-983a-abd539f49a0e.png)

Reply:
Argh I just noticed that Native American isn't listed in the raw data. It looks like "race" is blank for the entry with 6 cases - that could possibly be Native American. Time to slack Jason Lally!

Reply:
Re: unknown, I think you might be missing the case where Race = Unknown and Ethnicity = Not Hispanic or Latino. That would surface the 12 you're missing to get the right number of 348.

Re: Hispanic or Latino - I'm not sure. Is it possible that the dashboard is wrong? Or my understanding of what should fall into this category is wrong? It makes sense to me that only the first three rows of the raw data set should be added (and your logic does that), and that only adds up to 828. I'll revisit this later tonight or tomorrow.

Re: White - I didn't have as much time to look at this one, so I'll try to look at it again tomorrow, but I think you're getting a higher count b/c you're counting (Race: White) + (Ethnicity: Hispanic or Latino) as White, but I think it's already accounted for in Hispanic or Latino?

Reply:
I don't think the Hispanic or Latino discrepancy can be explained by this (the discrepancy seems too large), but just posting this disclaimer from the raw data set anyway: "Unknowns may become known over time as more information is gathered. Combinations of race and ethnicity resulting in 5 or fewer cases are dropped for privacy reasons.", in case it happens to be useful at some point.

Reply:
Some progress on this. Here's our data as of 5/17:
```
 "race_eth": {
        "Latinx_or_Hispanic": 881,
        "Asian": 263,
        "African_Amer": 105,
        "White": 323,
        "Pacific_Islander": 25,
        "Native_Amer": 8,
        "Multiple_Race": 10,
        "Other": 107,
        "Unknown": 346
      }
```
Here's what the dashboard is showing, updated 5/16:
![Screen Shot 2020-05-17 at 9 55 04 PM](https://user-images.githubusercontent.com/12928553/82175593-18121080-9889-11ea-93df-37420f15d262.png)

Here are the major changes I made to the logic:
  1. Include Race = 'Unknown', Ethnicity = 'Not Hispanic or Latino' in our 'Unknown'
2. Exclude Race = 'Other', Ethnicity = 'Hispanic or Latino' from our 'Other'
3. Exclude Race = 'White', Ethnicity = 'Hispanic or Latino' from our 'White
4. If Race was not reported, counted it the value as Race = 'Native American' (!!!)
This last one is because the dashboard has Native American, but the [raw data](https://data.sfgov.org/COVID-19/COVID-19-Cases-Summarized-by-Race-and-Ethnicity/vqqm-nsqg/data) doesn't, however, the one value in the raw data that doesn't have race reported looks to be very close to 'Native American'. 
I am adding this very heavy-handed interpretation to our metadata.
Thoughts? @kwonangela7 @Mr0grog

Reply:
> This last one is because the dashboard has Native American, but the raw data doesn't, however, the one value in the raw data that doesn't have race reported looks to be very close to 'Native American'.

Ahhhhhhh! Yeah, that makes sense, although, ugh, I definitely hear you on how making this assumption seems like a bad thing. Definitely worth asking the county, but I also wouldn’t block merging while waiting for that feedback (we can always make changes in the future when we hear back).

I also think 2 & 3 feel weird, or at least not what I’d expect based on how categories are named. and I think the best way to characterize this is that, even though our categories are cross-cutting (i.e., based on their name, someone could fit in more than one), we are ensuring every case only gets bucketed in *one* (so they sum up to 100% of cases, rather than > 100%).

**I don’t have a strong feel on whether doing this in order to match SF’s dashboard is good. If we do it, we should add a note about it to `meta_from_baypd`.** It might be good to add a big comment at the top of `get_race_eth_table()` that describes how cases are getting bucketed, similar to what you wrote here (that way someone can get a sense of it without needed to puzzle through the actual logic).

Reply:
So I've decided to assign each individual to one bucket. I've add a description in `meta_from_baypd` and in comments at the top of get_race_eth_table().
I think is what our logic was doing anyways, I've just made it more clear. In any case I'm getting the same numbers as before with this approach - close to what's on the SF dashboard. 
For comparison, here's a [snapshot of Sonoma county's race/eth table](
https://socoemergency.org/emergency/novel-coronavirus/coronavirus-cases/), 
![Screen Shot 2020-05-19 at 7 20 50 AM](https://user-images.githubusercontent.com/12928553/82338226-7f22e880-99a1-11ea-8ef3-e45884c68619.png)
It looks like they are bucketing each individual in one category, so the partition sums up to 100% of cases. Looks like they are only putting an individual in a race bucket if they are not Hispanic/Latino. Since Sonoma is doing it this way, and it looks like SF could be doing it this way, I'm inclined to do the same.